test/github-issues/9936/issue-9936.ts

```diff
@@ -0,0 +1,270 @@
+import "reflect-metadata"
+import { DataSource } from "../../../src"
+import {
+    closeTestingConnections,
+    createTestingConnections,
+    reloadTestingDatabases,
+} from "../../utils/test-utils"
+import { expect } from "chai"
+import { Region } from "./entity/Region"
+import { Country } from "./entity/Country"
+import { Province } from "./entity/Province"
+
+describe("github issues > #9936 Self-referencing relations in table inheritance cause duplicate join error with query strategy", () => {
```

Comment:
We are no longer creating tests for specific GitHub issues. Instead, new tests should be directly added to the test suite(s) for the functionality involved.

Suggest we move these tests to `test/functional/query-builder/single-table-inheritance`.